pulsar-impl/src/main/java/com/datastax/oss/pulsar/sink/codecs/PulsarCodecProvider.java

```diff
@@ -41,6 +41,12 @@ public class PulsarCodecProvider implements ConvertingCodecProvider {
             || externalJavaType.equals(GenericType.of(AbstractStruct.class)))) {
       return Optional.of(new StructToUDTCodec(codecFactory, (UserDefinedType) cqlType));
     }
+    // TODO: Generalize this part to handle Avro as well
+    // TODO: Check the behavior of the above if condition if the PulsarStruct wraps a native object.
```

Comment:
actually I tried some UDT scenarios and they work fine. Now we can mix UDT and Map/Array/etc. types in json objects and sink them without issues. The codec will not kick in if no schema (and therefore native object) where present. Will delete this TODO line when I address comments.

Reply:
can we add some e2e tests within this patch ?

Reply:
sure.